avm/res/compute/virtual-machine/main.bicep

```diff
@@ -1060,3 +1053,80 @@ type roleAssignmentType = {
   @description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+type osDiskType = {
+  @description('Optional. The disk name.')
+  name: string?
+
+  @description('Required. Specifies the size of an empty data disk in gigabytes.')
+  @maxValue(1023)
+  diskSizeGB: int
+
+  @description('Optional. Specifies how the virtual machine should be created.')
+  createOption: 'Attach' | 'Empty' | 'FromImage'?
+
+  @description('Optional. Specifies whether data disk should be deleted or detached upon VM deletion.')
+  deleteOption: 'Delete' | 'Detach'?
+
+  @description('Optional. Specifies the caching requirements.')
+  caching: 'None' | 'ReadOnly' | 'ReadWrite'?
+
+  @description('Required. The managed disk parameters.')
+  managedDisk: {
+    @description('Required. Specifies the storage account type for the managed disk.')
+    storageAccountType:
+      | 'PremiumV2_LRS'
+      | 'Premium_LRS'
+      | 'Premium_ZRS'
+      | 'StandardSSD_LRS'
+      | 'StandardSSD_ZRS'
+      | 'Standard_LRS'
+      | 'UltraSSD_LRS'
+
+    @description('Optional. Specifies the customer managed disk encryption set resource id for the managed disk.')
+    diskEncryptionSet: {
+      @description('Required. The resource ID of the disk encryption set.')
+      id: string
+    }?
+  }
+}
+
+type dataDisksType = {
+  @description('Optional. The disk name.')
+  name: string?
+
+  @description('Optional. Specifies the logical unit number of the data disk.')
+  lun: int?
+
+  @description('Required. Specifies the size of an empty data disk in gigabytes.')
+  @maxValue(1023)
+  diskSizeGB: int
+
+  @description('Optional. Specifies how the virtual machine should be created.')
+  createOption: 'Attach' | 'Empty' | 'FromImage'?
+
+  @description('Optional. Specifies whether data disk should be deleted or detached upon VM deletion.')
+  deleteOption: 'Delete' | 'Detach'?
+
+  @description('Optional. Specifies the caching requirements.')
+  caching: 'None' | 'ReadOnly' | 'ReadWrite'?
+
+  @description('Required. The managed disk parameters.')
+  managedDisk: {
+    @description('Required. Specifies the storage account type for the managed disk.')
+    storageAccountType:
+      | 'PremiumV2_LRS'
+      | 'Premium_LRS'
+      | 'Premium_ZRS'
+      | 'StandardSSD_LRS'
+      | 'StandardSSD_ZRS'
+      | 'Standard_LRS'
+      | 'UltraSSD_LRS'
+
+    @description('Optional. Specifies the customer managed disk encryption set resource id for the managed disk.')
+    diskEncryptionSet: {
```

Comment:
Usually, for cases where only a single ID entry exists, we'd opt to change the input object to a `diskEncryptionSetResourceId` and create the object for the user. While it divers slightly from the RP, it's easier to use, and more descriptive (be it only because it explicitly calls for a resourceId and nothering else). Same for the osDisk. Thoughts?

Reply:
I agree. It saves the user some object complexity and makes it more straight-forward. Replaced this in the osDisk and dataDisk.